schematic/schemas/validation_rule_functions.py

```diff
@@ -0,0 +1,386 @@
+"""This module contains functions for interacting with validation rules for JSON Schema creation"""
+
+from dataclasses import dataclass
+from typing import Optional
+from schematic.schemas.constants import JSONSchemaType, ValidationRuleName, RegexModule
+
+
+@dataclass
+class ValidationRule:
+    """
+    This class represents a validation rule to be used for creating JSON Schemas
+
+    Attributes:
+        name: The name of the validation rule
+        js_type: The JSON Schema type this rule indicates
+        is_type_rule: Whether or not this rule directly corresponds to a JSON Schema type
+        incompatible_rules: Other validation rules this rule can not be paired with
+        parameters: Parameters for the validation rule that need to be collected for the JSON Schema
+    """
+
+    name: ValidationRuleName
+    js_type: Optional[JSONSchemaType]
+    is_type_rule: bool
+    incompatible_rules: list[ValidationRuleName]
+    parameters: Optional[list[str]] = None
+
+
+# A List of validation rules currently collected to make a JSON Schema
+_VALIDATION_RULES = [
+    ValidationRule(
+        name=ValidationRuleName.LIST,
```

Comment:
why here `LIST` does not map to array: https://json-schema.org/understanding-json-schema/reference/type#type-specific-keywords? is it because `array` is not a supported key word in synapse?

Reply:
Great question! You are correct about array. See this design doc: